src/main/java/net/coderbot/iris/Iris.java

```diff
@@ -370,21 +371,17 @@ private static void destroyEverything() {
 		}
 	}
 
-	public static DimensionId lastDimension = DimensionId.OVERWORLD;
+	public static int lastDimension = 0;
 
-	public static DimensionId getCurrentDimension() {
+	public static int getCurrentDimension() {
 		ClientWorld world = MinecraftClient.getInstance().world;
 
 		if (world != null) {
 			RegistryKey<World> worldRegistryKey = world.getRegistryKey();
 
-			if (worldRegistryKey.equals(World.END)) {
-				return DimensionId.END;
-			} else if (worldRegistryKey.equals(World.NETHER)) {
-				return DimensionId.NETHER;
-			} else {
-				return DimensionId.OVERWORLD;
-			}
+			Identifier dimensionId = worldRegistryKey.getValue();
+
+			return getCurrentPack().get().getIdMap().getDimensionIdMap().getOrDefault(dimensionId, 0);
```

Comment:
Does anyone have a way this could be improved, this is extremely concerning.

Reply:
This is incorrect. An unrecognized dimension should not map to the world0 folder, it should map to the root shaders directory.